### DISCUSSION TITLE
Add support for NER formats that have XML tags using ENAMEX

### DISCUSSION THREAD

**Discussion by afshinrahimi** [explosion/spaCy#5407] [New Features & Project Ideas]
## Feature description
Add support to convert labelled NER files to Spacy format.
One example of such files is:

```
<ENAMEX TYPE="LOCATION">Italy</ENAMEX>'s business world was rocked by
the announcement <TIMEX TYPE="DATE">last Thursday</TIMEX> that Mr.
<ENAMEX TYPE=„PERSON">Verdi</ENAMEX> would leave his job as vicepresident of <ENAMEX TYPE="ORGANIZATION">Music Masters of Milan, Inc</ENAMEX> to become operations director of <ENAMEX TYPE="ORGANIZATION">Arthur Andersen</ENAMEX>.
```

## Could the feature be a [custom component](https://spacy.io/usage/processing-pipelines#custom-components) or [spaCy plugin](https://spacy.io/universe)?
If so, we will tag it as [`project idea`](https://github.com/explosion/spaCy/labels/project%20idea) so other users can take it on.

**Comment by adrianeboyd** [comment]
This could be implemented as an additional converter for the `spacy convert` CLI like the existing converters here: https://github.com/explosion/spaCy/tree/master/spacy/cli/converters

As an alternative, there may be existing converters that convert this format to one of the NER formats supported by spacy already. There are examples of the supported formats here: https://github.com/explosion/spaCy/tree/master/examples/training/ner_example_data

**Comment by afshinrahimi** [comment]
I have written a piece of code that works for me (not tested it on other datasets) in this repo https://github.com/afshinrahimi/markup2bio4ner
Two files, one for converting cases where you have` <enamex type="location">Italy</enamex>` and another one for when you have `<location>Italy</location>`.

**Comment by kevinlu1248** [comment]
I made something similar to @afshinrahimi at https://github.com/kevinlu1248/researchy-api but it's mainly for parsing HTML first and then annotating it with XML tags.

**Comment by svlandeg** [comment]
In general, we try to focus the main library on the most common use-cases/data formats, and we don't want to clutter the `convert` command too much. If you have written a convertor that should be generic enough for various use-cases, we could definitely add it to the [spacy universe](https://spacy.io/universe) though. If that's the case, please feel free to open a PR with the specifics of the plugin!

FYI - from v.3 onwards, the data format will work a bit differently, and the standard format will be `.spacy` files created by serializing `DocBin`, allowing us to standardize on the `Doc` format more.
